src/Mod/Measure/Gui/TaskMeasure.cpp

```diff
@@ -174,85 +174,36 @@ void TaskMeasure::enableAnnotateButton(bool state)
     btn->setEnabled(state);
 }
 
-void TaskMeasure::setMeasureObject(Measure::MeasureBase* obj)
-{
-    _mMeasureObject = obj;
-}
-
 
-App::DocumentObject* TaskMeasure::createObject(const App::MeasureType* measureType)
+Measure::MeasureBase* TaskMeasure::createObject(const App::MeasureType* measureType)
 {
-    auto measureClass =
-        measureType->isPython ? "Measure::MeasurePython" : measureType->measureObject;
-    auto type = Base::Type::getTypeIfDerivedFrom(measureClass.c_str(),
-                                                 App::DocumentObject::getClassTypeId(),
-                                                 true);
-
-    if (type.isBad()) {
+    App::Document* doc = App::GetApplication().getActiveDocument();
+    if (!doc) {
         return nullptr;
     }
 
-    _mMeasureObject = static_cast<Measure::MeasureBase*>(type.createInstance());
-
-    // Create an instance of the python measure class, the classe's
-    // initializer sets the object as proxy
     if (measureType->isPython) {
         Base::PyGILStateLocker lock;
-        Py::Tuple args(1);
-        args.setItem(0, Py::asObject(_mMeasureObject->getPyObject()));
-        PyObject* result = PyObject_CallObject(measureType->pythonClass, args.ptr());
-        Py_XDECREF(result);
-    }
-
-    return static_cast<App::DocumentObject*>(_mMeasureObject);
-}
+        auto pyMeasureClass = measureType->pythonClass;
 
+        // Create a MeasurePython instance
+        auto featurePython = doc->addObject("Measure::MeasurePython", measureType->label.c_str());
+        _mMeasureObject = (Measure::MeasureBase*)featurePython;
```

Comment:
Do not use C-style casts. They are considered harmful because they can act like `static_cast`, `reinterpret_cast` or `const_cast` or a combination of them and you don't know which of the casts are actually performed.

Use a `dynamic_cast` or `static_cast` here.